### PR TITLE
Master project subtask timesheet flow hbh

### PR DESCRIPTION
**Pull request by hiralvbhavsar**
Description of the issue/feature this PR addresses:

Task: https://www.odoo.com/web?#id=32400&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.f78d3892989948bd3ff8d41246a32895

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/project/models/project.py [hunk 1]
```diff
@@ -438,6 +446,17 @@ def copy(self, default=None):
             default['remaining_hours'] = self.planned_hours
         return super(Task, self).copy(default)
 
+    @api.multi
+    def _get_subtask_count(self):
```
Comment (jem-odoo) [inline addons/project/models/project.py]: Please rename it into `_compute_subtask_count` (and check if it's overriden somewhere else).

File: addons/web/static/src/js/views/kanban/kanban_column.js [hunk 2]
```diff
@@ -141,7 +142,7 @@ var KanbanColumn = Widget.extend({
         }
         var self = this;
         var width = this.records.length ? this.records[0].$el.innerWidth() : this.$el.width() - 8;
-        this.quickCreateWidget = new RecordQuickCreate(this, width);
+        this.quickCreateWidget = new RecordQuickCreate(this, width, {default_name: this.data.getContext().default_name});
```
Comment (jem-odoo) [inline addons/web/static/src/js/views/kanban/kanban_column.js]: Isn't it going to crash if there is no 'default_name' ?
I think we should have @ged-odoo advice here.

File: addons/web/static/src/js/views/kanban/kanban_quick_create.js [hunk 3]
```diff
@@ -117,16 +117,20 @@ var RecordQuickCreate = AbstractQuickCreate.extend({
     /**
      * @override
      */
-    init: function (parent, width) {
+    init: function (parent, width, options) {
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_quick_create.js]: you should put the 'width' argument into the options: init: function (parent, options)

Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_quick_create.js]: missing docstring: parent and options are not doumented

File: addons/web/static/src/js/views/kanban/kanban_quick_create.js [hunk 4]
```diff
@@ -117,16 +117,20 @@ var RecordQuickCreate = AbstractQuickCreate.extend({
...
         this._super.apply(this, arguments);
         this.width = width;
+        if (options) {
+            this.default_name = options.default_name;
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_quick_create.js]: variables should be camelcased (in javascript): defaultName

File: addons/web/static/src/js/views/kanban/kanban_column.js (hunk 2 above)
Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_column.js]: @jem-odoo it's ok here, the kanban quick create has been properly adapted

Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_column.js]: but it should not be called default_name.  It should be defaultName

**Comment by dreispt** [comment]
So, if we want to use SubTasks, we are forced to have Timesheets?

**Comment by Icallhimtest** [comment]
@dreispt No, that was the previous behaviour. The idea is that after this commit you will be able to use subtasks without timesheets.

File: addons/web/static/src/js/views/kanban/kanban_quick_create.js [hunk 5]
```diff
@@ -116,17 +116,23 @@ var RecordQuickCreate = AbstractQuickCreate.extend({
...
+     * @param {Object} options
+     * @param {string} options.width defines the element's width
+     * @param {string} options.defaultName the record's default name
      */
-    init: function (parent, width) {
+    init: function (parent, options) {
         this._super.apply(this, arguments);
-        this.width = width;
+        this.width = options.width
+        this.defaultName = options.defaultName;
```
Comment (Icallhimtest) [inline addons/web/static/src/js/views/kanban/kanban_quick_create.js]: @ged-odoo docstring and options good for you ?

File: addons/web/static/src/js/views/kanban/kanban_quick_create.js [hunk 6]
```diff
@@ -116,17 +116,23 @@ var RecordQuickCreate = AbstractQuickCreate.extend({
     template: "KanbanView.QuickCreate",
     /**
      * @override
+     * @param {Widget} parent
+     * @param {Object} options
+     * @param {string} options.width defines the element's width
+     * @param {string} options.defaultName the record's default name
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_quick_create.js]: +     * @param {string} [options.defaultName] the record's default name

to mark a parameter as optional, it should be enclosed in brackets

File: addons/web/static/src/js/views/kanban/kanban_quick_create.js [hunk 7]
```diff
@@ -116,17 +116,23 @@ var RecordQuickCreate = AbstractQuickCreate.extend({
     template: "KanbanView.QuickCreate",
     /**
      * @override
+     * @param {Widget} parent
+     * @param {Object} options
+     * @param {string} options.width defines the element's width
```
Comment (ged-odoo) [inline addons/web/static/src/js/views/kanban/kanban_quick_create.js]: width is a string?

**Pull request merged**